src/Sylius/Behat/Context/Setup/ProductContext.php

```diff
@@ -170,6 +188,88 @@ public function productVariantBelongsToTaxCategory(
     }
 
     /**
+     * @Given /^(this product) has ([^"]+) attribute "([^"]+)" with value "([^"]+)"$/
+     */
+    public function thisProductHasAttributeWithValue(ProductInterface $product, $productAttributeType, $productAttributeName, $value)
+    {
+        $this->createProductAttribute($productAttributeType,$productAttributeName);
+        $this->createProductAttributeValue($value);
+        $product->addAttribute($this->sharedStorage->get('product_attribute_value'));
```

Comment:
This seems strange to me, shouldn't we get attribute here, create attribute value and add it to the product? Why the need for shared storage here?